cmd/terramate/cli/cloud.go

```diff
@@ -484,7 +485,15 @@ func (c *cli) detectCloudMetadata() {
 			Msg("failed to retrieve commit information from GitHub API")
 	}
 
-	if pull, found, err := getGithubPR(githubClient, r.Owner, r.Name, headCommit); err == nil {
+	var pullRequest *github.PullRequest
+	pullRequest, err = tmgithub.GetEventPR()
+	if err != nil {
+		printer.Stderr.WarnWithDetails(
+			sprintf("unable to read pull_request from GITHUB_EVENT_PATH, falling back to github API"),
+			err)
+	}
+
+	if pull, found, err := getGithubPR(githubClient, r.Owner, r.Name, headCommit); err == nil && pullRequest == nil {
```

Comment:
Am I missing something or the retrieved `pullRequest` from the event file is not being used and GH API call is done anyway? I would expect this line to be in the `else` if the `tmgithub.GetEventPR()` fails.

Reply:
You are right, that's broken. I'll make this PR a draft for now and make changes to parse the file only once.

Reply:
fixed